src/parser/jobs/sch/ClericStance.js

```diff
@@ -0,0 +1,193 @@
+import ACTIONS, {getAction} from 'data/ACTIONS'
+import Module from 'parser/core/Module'
+import {SEVERITY, TieredSuggestion} from 'parser/core/modules/Suggestions'
+import {Accordion, Message, Icon} from 'semantic-ui-react'
+import Rotation from 'components/ui/Rotation'
+import STATUSES from 'data/STATUSES'
+
+// all these suggestions for CS are ultimately trivial :notlikeblob:
+const defaultSeverityTiers = {
+	1: SEVERITY.MINOR,
+}
+
+// list of things to track for CS usage.
+// can override the suggestions from the default template here
+const EXPECTED_CASTS = [
+	{...getAction(ACTIONS.CLERIC_STANCE.id), name: 'GCD', count: 6}, // track GCDs via the CS id
+	{...getAction(ACTIONS.BIO_II.id), count: 1},
+	{...getAction(ACTIONS.MIASMA.id), count: 1},
+	{...getAction(ACTIONS.SHADOW_FLARE.id),
+		count: 1,
+		content: <Fragment>
+			Try to land a <ActionLink {...ACTIONS.SHADOW_FLARE} /> during <ActionLink {...ACTIONS.CLERIC_STANCE} /> if both will be available at the same time.
+			Avoid casting <ActionLink {...ACTIONS.SHADOW_FLARE} /> right before  <ActionLink {...ACTIONS.CLERIC_STANCE} />
+		</Fragment>},
+]
+
+export default class ClericStance extends Module {
+	static displayOrder = 99;
+	static handle = 'clericstance';
+	static dependencies = [
+		'suggestions',
+		'cooldowns',
+		'gcd',
+	];
+
+	static title = 'Cleric Stance Usage'
+
+	constructor(...args) {
+		super(...args)
+
+		this.addHook('cast', {by: 'player'}, this._onCast)
+		this.addHook('removebuff', {to: 'player', abilityId: [STATUSES.CLERIC_STANCE.id]}, this._onEnd)
+		this.addHook('complete', this._onComplete)
+	}
+
+	_buffStart = null
+	_buffRotations = {}
+	_potentialShadowFlareUsages = [];
+
+	_onCast(event) {
+		const actionId = event.ability.guid
+
+		if (actionId === ACTIONS.ATTACK.id) {
+			return
+		}
+
+		if (actionId === ACTIONS.CLERIC_STANCE.id) {
+			this._buffStart = event.timestamp
+			this._buffRotations[this._buffStart] = []
+
+			if (this._canShadowFlareBeUsed()) {
+				this._potentialShadowFlareUsages.push(this._buffStart)
+			}
+
+		}
+
+		if (this._buffStart) {
+			this._buffRotations[this._buffStart].push(event)
+		}
+	}
+
+	_canShadowFlareBeUsed() {
+		const {id, cooldown} = getAction(ACTIONS.SHADOW_FLARE.id)
+
+		const cooldownRemaining = this.cooldowns.getCooldownRemaining(id)
+		if (cooldownRemaining - (STATUSES.CLERIC_STANCE.duration * 1000) < this.gcd.getEstimate()) {
+			// cooldown will come off before CS duration expires
+			// this probably won't be accurate if the CS duration gets extended, idk.
+			return true
+		} if ((cooldown * 1000 - cooldownRemaining) < this.gcd.getEstimate()) {
+			// if it was recently used as far as 1 gcd before popping cleric,
+			// let's count it as a potential wasted usage
+			return true
+		}
+
+		return false
+	}
+
+	_onEnd() {
+		this._buffStart = null
+	}
+
+	_onComplete() {
+		EXPECTED_CASTS.forEach(({id, name, icon, count, why, content, tiers}) => {
+			const expected = Object.keys(this._buffRotations).reduce(
```

Comment:
Minor thing, that could be considered here, since we have now access to it (this is purely optional and for code cleanliness) would be using the lodash reduce, since that one supports objects right out of the gate.

```javascript
_.reduce(this._buffRotations, (total, rotation, timestamp) => [ ... ], 0)
```